x/posts/internal/types/poll_data.go

```diff
@@ -0,0 +1,269 @@
+package types
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// ---------------
+// --- PollData
+// ---------------
+
+// PollData contains the information of a poll that is associated to a post
+type PollData struct {
+	Title                 string      `json:"title"`                   // Describes what poll is about
+	ProvidedAnswers       PollAnswers `json:"provided_answers"`        // Lists of answers provided by the creator
+	EndDate               time.Time   `json:"end_date"`                // RFC3339 date at which the poll will no longer accept new answers
+	Open                  bool        `json:"open"`                    // Tells if the poll is still accepting answers
+	AllowsMultipleAnswers bool        `json:"allows_multiple_answers"` // Tells if the poll is a single or multiple answers one
+	AllowsAnswerEdits     bool        `json:"allows_answer_edits"`     // Tells if the poll allows answer edits
+}
+
+func NewPollData(title string, endDate time.Time, providedAnswers PollAnswers, open, allowMultipleAnswers, allowsAnswerEdits bool) *PollData {
+	return &PollData{
+		Title:                 title,
+		EndDate:               endDate,
+		ProvidedAnswers:       providedAnswers,
+		Open:                  open,
+		AllowsMultipleAnswers: allowMultipleAnswers,
+		AllowsAnswerEdits:     allowsAnswerEdits,
+	}
+}
+
+// String implements fmt.Stringer
+func (pd PollData) String() string {
+	bytes, err := json.Marshal(&pd)
```

Comment:
I would avoid using `json.Marshal` inside the `String` method as it may be counter intuitive in specific cases (like the CLI output). Instead I would prefer a more simple approach.